src/main/java/org/elasticsearch/script/ScriptParams.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import java.util.Map;
+
+import static org.elasticsearch.script.ScriptService.ScriptType;
+
+/**
+ * ScriptParams holds all the parameters necessary to compile or find in cache and then execute a script.
+ */
+public class ScriptParams {
```

Comment:
Script is used in a number of places throughout these classes, so I thought it made sense to differentiate from that with ScriptParams -- all the parameters necessary to run a script.  However, I will change it back since both of you prefer the name to be just "Script."